Aptoide/aptoide-client-v8
# [feature/ASV-1339] Intent for retrieving wallet address

**What does this PR do?**

   This PR aims to change the way we retrieve the wallet address. Now the wallet address is retrieved by sending an intent to the wallet app and fallbacks to the copy paste method if it doesn't work. Additionally, the fragment/presenter architecture was restructured to follow the rest of the project, namely BaseDialogView and BaseDialogFragment were created to accomodate RX lifecycle events and fragment scope injections.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] ClaimPromotionDialogFragment.java
- [ ] ClaimPromotionDialogPresenter.kt

**How should this be manually tested?**

  Check the different use cases and if everything is working correctly, namely:
  1) If the wallet app is installed and supports the intent, does it retrieve the wallet address correctly?
  2) If the wallet app is either uninstalled or the version doesn't support the intent, does it fallback correctly to the copy paste method?

**What are the relevant tickets?**

  [ASV-1339](https://aptoide.atlassian.net/browse/ASV-1339)

**Code Review Checklist**
